### PR TITLE
adding terraform scripts

### PR DESCRIPTION
**Pull request by micson-us**
broke down terraform scripts

### CODE REVIEW

File: aws_xis_taskdefinition.tf [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# creates the XIA task definition. container definition is defined in the *.json file
+resource "aws_ecs_task_definition" "dau_xis_task_definition" {
+  family                = "dau_xis_task_definition"
```
Comment (mattdrayer) [inline aws_xis_taskdefinition.tf:3]: Can remove `dau_` prefix from here, I think.

Comment (mattdrayer) [inline aws_xis_taskdefinition.tf:3]: remove `dau_`

File: aws_xse_taskdefinition.tf [hunk 2]
```diff
@@ -0,0 +1,18 @@
+# creates the XIA task definition. container definition is defined in the service.json file
```
Comment (mattdrayer) [inline aws_xse_taskdefinition.tf:1]: XIA -> XSE

File: aws_xss_ecs_cluster.tf [hunk 3]
```diff
@@ -0,0 +1,26 @@
+# creates the ECS cluster for XSS
+resource "aws_ecs_cluster" "dau_xss_cluster" {
```
Comment (mattdrayer) [inline aws_xss_ecs_cluster.tf:2]: Remove `dau_`

File: aws_xss_ecs_cluster.tf [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+  name = "dau_xss_cluster"
```
Comment (mattdrayer) [inline aws_xss_ecs_cluster.tf:3]: Remove `dau_`

File: aws_xss_ecs_cluster.tf [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+}
+
+# creates the ECS service
+resource "aws_ecs_service" "dau_xss_service" {
```
Comment (mattdrayer) [inline aws_xss_ecs_cluster.tf:7]: remove `dau_`

File: aws_xss_ecs_cluster.tf [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+  name            = "dau_xia"
```
Comment (mattdrayer) [inline aws_xss_ecs_cluster.tf:8]: -> `xss_ecs_service` -- does this need to align with the resource name?  seems like that would make sense.

File: aws_xss_ecs_cluster.tf [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+  cluster         = aws_ecs_cluster.dau_xss_cluster.id
```
Comment (mattdrayer) [inline aws_xss_ecs_cluster.tf:9]: remove `dau_`

File: aws_xss_ecs_cluster.tf [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+  task_definition = aws_ecs_task_definition.dau_xss_task_definition.arn
```
Comment (mattdrayer) [inline aws_xss_ecs_cluster.tf:10]: remove `dau_`

File: aws_xss_taskdefinition.tf [hunk 9]
```diff
@@ -0,0 +1,18 @@
+# creates the XSS task definition. container definition is defined in the service.json file
```
Comment (mattdrayer) [inline aws_xss_taskdefinition.tf:1]: recommend renaming file to `aws_xss_task_definition.tf`

File: aws_xss_taskdefinition.tf [hunk 10]
```diff
@@ -0,0 +1,18 @@
...
+resource "aws_ecs_task_definition" "xss_ecs_task_definition" {
+  family                = "xss_ecs_task_definition"
+  container_definitions = file("task-definitions/dauxssimages.json")
```
Comment (mattdrayer) [inline aws_xss_taskdefinition.tf:4]: remove `dau` -> `xss_images.json`

File: iamrole.tf [hunk 11]
```diff
@@ -0,0 +1,64 @@
+resource "aws_iam_role" "ecs_task_role" {
```
Comment (mattdrayer) [inline iamrole.tf:1]: rename file to `aws_iam_role.tf`, yes?

File: iamrole.tf [hunk 12]
```diff
@@ -0,0 +1,64 @@
...
+      },
+      "Effect": "Allow",
+      "Sid": ""
+    }
+  ]
+}
+EOF
+
+  tags = {
+    Name = "ecs role"
```
Comment (mattdrayer) [inline iamrole.tf:21]: Not sure if it matters, but should `Name` be lowercase or in the current case format?

File: task-definitions/dauxiaimages.json [hunk 13]
```diff
@@ -0,0 +1,54 @@
+[
```
Comment (mattdrayer) [inline task-definitions/dauxiaimages.json:1]: Rename file to xia_dau_images.json

File: task-definitions/dauxiaimages.json [hunk 14]
```diff
@@ -0,0 +1,54 @@
...
+   {
+      "name":"xia-app",
+      "image":"jmdiouf/openlxp-xia-dau:latest",
```
Comment (mattdrayer) [inline task-definitions/dauxiaimages.json:4]: at some point we'll need to reference a more official image than Mac's.

File: task-definitions/dauxisimages.json [hunk 15]
```diff
@@ -0,0 +1,54 @@
+[
```
Comment (mattdrayer) [inline task-definitions/dauxisimages.json:1]: rename to `xis_images.json`

File: task-definitions/dauxssimages.json [hunk 16]
```diff
@@ -0,0 +1,54 @@
+[
```
Comment (mattdrayer) [inline task-definitions/dauxssimages.json:1]: rename to `xss_images.json`

File: task-definitions/dauxssimages.json [hunk 17]
```diff
@@ -0,0 +1,54 @@
...
+   {
+      "name":"xss-app",
+      "image":"jmdiouf/openlxp-xia-dau:latest",
```
Comment (mattdrayer) [inline task-definitions/dauxssimages.json:4]: I think we need to reference a different image here -- perhaps `jmdiouf/openlxp-xss:latest`

File: task-definitions/dauxisimages.json [hunk 18]
```diff
@@ -0,0 +1,54 @@
...
+   {
+      "name":"xis-app",
+      "image":"jmdiouf/openlxp-xia-dau:latest",
```
Comment (mattdrayer) [inline task-definitions/dauxisimages.json:4]: I think we need to reference a different image -- `jmdiouf/openlxp-xis:latest`, yes?

File: task-definitions/xse_elasticsearchimage.json [hunk 19]
```diff
@@ -0,0 +1,17 @@
+[
+  {
```
Comment (mattdrayer) [inline task-definitions/xse_elasticsearchimage.json:2]: Recommend renaming file to `xse_images.json`, or possibly `xse_images_elasticsearch.json`

File: variables.tf [hunk 20]
```diff
@@ -0,0 +1,25 @@
+variable "profile" {
```
Comment (mattdrayer) [inline variables.tf:1]: For my own understanding, why would we add an entry to this file, versus embedding the information directly in the TF script. Is it for reuse across scripts?  Operator-specified overrides?

**Review by mattdrayer — CHANGES_REQUESTED** [review]
This is looking really great!  Mainly I just pointed out some inconsistencies and possibly invalid references -- we should be able to finish this PR up soon!

File: task-definitions/dauxssimages.json (hunk 17 above)
Comment (micson-us) [reply task-definitions/dauxssimages.json:4]: Hey Matt - I agree. I just have this as a place holder until we have specific component images created and pushed.

File: variables.tf (hunk 20 above)
Comment (micson-us) [reply variables.tf:1]: Yup it is for reuse across scripts however, as I made updates to the scripts, I do not think I am calling these variables. I can plan on removing this file or adding more redundant entries to the variables and use that across scripts.

File: aws_xis_taskdefinition.tf (hunk 1 above)
Comment (micson-us) [reply aws_xis_taskdefinition.tf:3]: Thanks for catching these!! made copies of the original XIA script and missed updating those

File: variables.tf (hunk 20 above)
Comment (mattdrayer) [reply variables.tf:1]: ok sounds good -- I like the idea of a variables file for information that is shared across scripts.  It's easy to go overboard and make everything a shared variable even if it's not actually shared, so we'll just want to be sure we only generalize things that are truly shared.

File: aws_xia_task_definition.tf [hunk 21]
```diff
@@ -0,0 +1,18 @@
+# creates the XIA task definition. container definition is defined in the *.json file
+resource "aws_ecs_task_definition" "xia_dau_task_definition" {
+  family                = "xia_dau_task_definition"
+  container_definitions = file("task-definitions/dau_xia_images.json")
```
Comment (mattdrayer) [inline aws_xia_task_definition.tf:4]: `xia_dau_images.json`

File: aws_xia_task_definition.tf [hunk 22]
```diff
@@ -0,0 +1,18 @@
+# creates the XIA task definition. container definition is defined in the *.json file
+resource "aws_ecs_task_definition" "xia_dau_task_definition" {
+  family                = "xia_dau_task_definition"
```
Comment (mattdrayer) [inline aws_xia_task_definition.tf:3]: It seems we have some XIA-related files that have to do with all XIAs, and some that have to do with the DAU XIA.  Is the goal to have all XIA AWS resources managed within the same set of XIA files, or will each XIA have its own files.  For example, should the name of this file be `aws_xia_task_definition.tf` or `aws_xia_dau_task_definition.tf`?

**Review by mattdrayer — APPROVED** [review]
One comment regarding general XIA scripts versus specific XIA scripts (eg, DAU XIA vs. EDX XIA).  If we can resolve that then I am good with merging what we have to `main`.

File: aws_xia_task_definition.tf (hunk 22 above)
Comment (micson-us) [reply aws_xia_task_definition.tf:3]: I think they should have separate task definitions so the name should be 'aws_xia_dau_task_definition.tf', 'aws_xia_jko_task_definition.tf', 'aws_xia_edx_task_definition.tf', etc. Thanks for catching that!

File: aws_xse_taskdefinition.tf (hunk 2 above)
Comment (mattdrayer) [inline aws_xse_taskdefinition.tf:1]: file name -> `aws_xse_task_definition.tf`

File: aws_xss_taskdefinition.tf (hunk 9 above)
Comment (mattdrayer) [inline aws_xss_taskdefinition.tf:1]: file name -> `aws_xss_task_definition.tf`

Comment (mattdrayer) [reply aws_xss_taskdefinition.tf:1]: Oh, wait, I'm seeing that file name above -- what's the difference between these two files?

File: task-definitions/dau_xia_images.json [hunk 23]
```diff
@@ -0,0 +1,54 @@
+[
```
Comment (mattdrayer) [inline task-definitions/dau_xia_images.json:1]: file name -> `xia_dau_images.json`

**Review by mattdrayer — APPROVED** [review]

**Review by jdiouf — APPROVED** [review]

**Pull request merged**